File: LeetCode/1078-occurrences-after-bigram.go
package main

import "fmt"
import "strings"

func findOcurrences(text string, first string, second string) []string {
	var res []string
	list := strings.Split(text, " ")
	size := len(list)
	if size < 3 {
		return res
	}
	for i := 0; i < size-2; i++ {
		if list[i] == first && list[i+1] == second {
			res = append(res, list[i+2])
		}
	}
	return res
}

func main() {
	fmt.Println(findOcurrences("alice is a good girl she is a good student",
		"a",
		"good"))
}

File: LeetCode/1078-occurrences-after-bigram.rs
impl Solution {
    pub fn find_ocurrences(text: String, first: String, second: String) -> Vec<String> {
        let mut ret = Vec::new();
        let list:  Vec<&str> = text.split_whitespace().collect();
        if list.len() < 3 {
            return ret
        }
        for i in (0..list.len()-2) {
            if list[i] == first && list[i+1] == second {
                ret.push(list[i+2].to_string())
            }
        }
        return ret
    }
}
